### PR TITLE
Fix revisit with one question

### PR DESCRIPTION
**Pull request by ankitjavalkar**

### CODE REVIEW

File: yaksh/test_models.py [hunk 1]
```diff
@@ -507,12 +507,19 @@ def setUpClass(self):
...
+        )
+        self.qtn_paper_with_single_question.save()
+
         self.questions = Question.objects.all()[0:3]
         self.start_time = timezone.now()
         self.end_time = self.start_time + timedelta(minutes=20)
         self.question1 = self.questions[0]
         self.question2 = self.questions[1]
         self.question3 = self.questions[2]
+        self.question4 = Question.objects.all()[3]
```
[the remark this answers is not in the document]
Comment (ankitjavalkar) [reply yaksh/test_models.py]: Please see line 539 ```self.questions``` is used in another context in different places, changing this just for consistency seems inappropriate.

[the remark this answers is not in the document]
Comment (prathamesh920) [reply yaksh/test_models.py]: okay, but ```Question.objects.all()``` is called twice at same place looks redundant. I am fine with this but could have been better if we had a single variable holding all the questions and then extracting questions from that variable as per the requirement.
I mean, for example
```
all_questions = Question.objects.all()
self.questions = all_questions[:3]
self.question1 = all_questions[0]
```
and so on

File: yaksh/test_models.py [hunk 2]
```diff
@@ -555,6 +562,32 @@ def setUpClass(self):
...
+            question_paper=self.question_paper,
+            start_time=self.start_time,
+            end_time=self.end_time,
+            user_ip=self.ip
+        )
+        self.attempted_papers = AnswerPaper.objects.filter(
+            question_paper=self.question_paper,
+            user=self.user
+        )
+        self.qtn_paper_with_single_question.fixed_questions.add(self.question4)
```
[the remark this answers is not in the document]
Comment (ankitjavalkar) [reply yaksh/test_models.py]: Fixed

File: yaksh/test_models.py [hunk 3]
```diff
@@ -614,6 +647,39 @@ def setUpClass(self):
             self.user2, self.ip, 1
         )
 
+    def test_returned_question_is_not_none(self):
+        # Test add_completed_question and next_question
+        # When all questions are answered
+
+        current_question = self.answerpaper_single_question.add_completed_question(
+            self.question1.id
```
[the remark this answers is not in the document]
Comment (ankitjavalkar) [reply yaksh/test_models.py]: Fixed

File: yaksh/test_models.py [hunk 4]
```diff
@@ -614,6 +647,39 @@ def setUpClass(self):
...
+        )
+
+        # Then
+        self.assertEqual(self.answerpaper_single_question.questions_left(), 0)
```
[the remark this answers is not in the document]
Comment (ankitjavalkar) [reply yaksh/test_models.py]: Fixed

File: yaksh/urls.py [hunk 5]
```diff
@@ -100,7 +100,7 @@
             views.regrade, name='regrade'),
     url(r'^manage/regrade/paper/(?P<course_id>\d+)/(?P<answerpaper_id>\d+)/$',
             views.regrade, name='regrade'),
-    url(r'^manage/(?P<mode>[\w\-]+)/(?P<quiz_id>\d+)/$', views.test_quiz),
+    url(r'^manage/(?:godmode|usermode)/(?P<quiz_id>\d+)/$', views.test_quiz),
```
Comment (prathamesh920) [inline yaksh/urls.py]: This will not work!
Because "mode" argument is required by the views.
Kindly fix this and test once.

**Comment by prathamesh920** [comment]
LGTM

**Pull request merged**